opentelemetry-sdk/tests/test_util.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+import unittest
+
+from opentelemetry.sdk.util import BoundedDict, BoundedList
+
+
+class TestBoundedList(unittest.TestCase):
+    base = [52, 36, 53, 29, 54, 99, 56, 48, 22, 35, 21, 65, 10, 95, 42, 60]
+
+    def test_negative_maxlen(self):
+        with self.assertRaises(ValueError):
+            BoundedList(-1)
+
+    def test_from_seq(self):
+        list_len = len(TestBoundedList.base)
+        blist = BoundedList.from_seq(list_len, TestBoundedList.base)
+
+        self.assertEqual(len(blist), list_len)
+
+        for idx in range(list_len):
+            self.assertEqual(blist[idx], TestBoundedList.base[idx])
+
+        # sequence too big
+        with self.assertRaises(ValueError):
+            BoundedList.from_seq(list_len / 2, TestBoundedList.base)
```

Comment:
You could also add an assertion that modifying base does not modify the BoundedList or vice versa